src/source/Sdp/Deserialize.c

```diff
@@ -74,74 +96,80 @@ STATUS deserializeSessionDescription(PSessionDescription pSessionDescription, PC
 {
     ENTERS();
     STATUS retStatus = STATUS_SUCCESS;
-    PCHAR curr, tail, next;
-    UINT32 lineLen;
-    CHK(sdpBytes != NULL, STATUS_SESSION_DESCRIPTION_INVALID_SESSION_DESCRIPTION);
+    SdpResult_t sdpResult = SDP_RESULT_OK;
+    SdpDeserializerContext_t ctx;
+    CHAR* pValue;
+    SIZE_T valueLength;
+    UINT8 type;
+    SIZE_T minLength;
 
-    curr = sdpBytes;
-    tail = sdpBytes + STRLEN(sdpBytes);
+    CHK(sdpBytes != NULL, STATUS_SESSION_DESCRIPTION_INVALID_SESSION_DESCRIPTION);
 
-    while ((next = STRNCHR(curr, tail - curr, '\n')) != NULL) {
-        lineLen = (UINT32) (next - curr);
+    sdpResult = SdpDeserializer_Init(&ctx, sdpBytes, STRLEN(sdpBytes));
+    CHK(sdpResult == SDP_RESULT_OK, sdpResult);
 
-        if (lineLen > 0 && curr[lineLen - 1] == '\r') {
-            lineLen--;
-        }
+    for (; sdpResult == SDP_RESULT_OK;) {
```

Comment:
Using both SDP_RESULT_OK (sdpResult) and STATUS_SUCCESS (retStatus) can be a bit confusing, should we stick to using only one of them throughout, as i think the value is same.

Reply:
Even though they're same value, SDP_RESULT_OK is returned from SDP library and STATUS_SUCCESS is used in whole webrtc sdk. IMHO we shouldn't mix them.

Reply:
Should we have a mapping to map STUN library errors to webrtc SDK error ? I am fine with keeping it as it as you suggested, but give it a thought if it can be improved.

Reply:
Updated